pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -10,10 +10,16 @@ import (
 )
 
 // NewRouteWeightedCluster takes a route and weighted cluster and returns a *RouteWeightedCluster
-func NewRouteWeightedCluster(route HTTPRouteMatch, weightedCluster service.WeightedCluster) *RouteWeightedClusters {
+func NewRouteWeightedCluster(route HTTPRouteMatch, weightedClusters []service.WeightedCluster) *RouteWeightedClusters {
```

Comment:
instead of a slice should we pass a set here ?

Reply:
It'd be nice if this were a slice bc then we can ensure that the type passed in is indeed service.WeightedCluster. If we pass in a set, there is some room for error bc it would be up to the package calling this method to ensure the type passed in is a set of service.WeightedCluster